### PR TITLE
NEW: SAMFire

### PR DESCRIPTION
**Pull request by to266**
An algorithm we came up with to significantly simplify fitting.

The paper is still being written, but in the mean time there may be things missing that I can't spot because I've been using this for the last 2 years....

Here is a quick example of usage:
https://gist.github.com/to266/4f8351998a0e9328d3e89a1d47745587

P.S.: I'll add more information about SAMFire once I have some more time. In the mean time I suppose a couple of eyes would not hurt

### CODE REVIEW

File: doc/user_guide/samfire.rst [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+
+.. versionadded:: 1.0
+    SAMFire
+
+SAMFire (Smart Adaptive Multi-dimensional Fitting) is an algorithm created to
+avoid the starting value (or local / false minima) problem, which often arises
+when fitting multi-dimensional datasets.
+
+The algorithm will be described in full when accompanying paper is published,
+however we make the implementation available before that.
```
Comment (tjof2) [inline doc/user_guide/samfire.rst]: "The algorithm will be described in full when the accompanying paper is published,
but we are making the implementation available now."
...or similar

File: doc/user_guide/samfire.rst [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+
+The idea
+--------
+
+The main idea of SAMFire is to change two things, when compared to the
+traditional way of fitting datasets with many dimensions in the navigation
+space:
```
Comment (tjof2) [inline doc/user_guide/samfire.rst]: "The main idea of SAMFire is to change two things compared to the..."

File: doc/user_guide/samfire.rst [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+ #. Calculate the pixel starting parameters from already fitted parts of the
+     dataset.
+
+Both of these aspects are linked one to another and are represented by two
+different strategy families that SAMFfire uses while operating.
+
+Strategies
+**********
+
+While operating, SAMFire uses a list of strategies to determine how to select
```
Comment (tjof2) [inline doc/user_guide/samfire.rst]: "During operation, SAMFire..." and "determine how to select THE next pixel"

File: doc/user_guide/samfire.rst [hunk 4]
```diff
@@ -0,0 +1,149 @@
...
+next pixel and estimate its starting parameters. Only one strategy is used at a
+time. Next strategy is activated when no new pixels are able to fit using the
```
Comment (tjof2) [inline doc/user_guide/samfire.rst]: How about "The next strategy is chosen from the list when no new pixels can be fitted with the current strategy."

**Comment by francisco-dlp** [comment]
The tests are not passing, maybe you forgot to add something to setup.py?

**Comment by francisco-dlp** [comment]
Also, could you update it with master?

**Comment by francisco-dlp** [comment]
Still failing

**Pull request merged**